--- /usr/lib/tmpfiles.d/opencryptoki.conf	2018-04-11 01:04:15.000000000 +0000
+++ /usr/lib/tmpfiles.d/opencryptoki.conf	2018-09-15 15:31:53.118712287 +0000
@@ -1,4 +1,4 @@
-D /var/lock/opencryptoki 0770 root pkcs11 -
-D /var/lock/opencryptoki/icsf 0770 root pkcs11 -
-D /var/lock/opencryptoki/swtok 0770 root pkcs11 -
-D /var/lock/opencryptoki/tpm 0770 root pkcs11 -
+D /run/lock/opencryptoki 0770 root pkcs11 -
+D /run/lock/opencryptoki/icsf 0770 root pkcs11 -
+D /run/lock/opencryptoki/swtok 0770 root pkcs11 -
+D /run/lock/opencryptoki/tpm 0770 root pkcs11 -
