#
# Workaround https://fedorahosted.org/freeipa/ticket/6518
#
--- /usr/lib/python2.7/site-packages/ipaserver/install/installutils.py	2018-09-18 15:47:56.000000000 +0000
+++ /usr/lib/python2.7/site-packages/ipaserver/install/installutils.py	2018-11-08 14:14:19.264503298 +0000
@@ -630,7 +630,7 @@
     # to avoid cases where we get IP address for an totally different name
     # but there is no way to do this using getaddrinfo parameters
     try:
-        addrinfos = socket.getaddrinfo(fqdn, None,
+        addrinfos = socket.getaddrinfo(fqdn.rstrip("."), None,
                                        socket.AF_UNSPEC, socket.SOCK_STREAM)
     except socket.error as ex:
         if ex.errno == socket.EAI_NODATA or ex.errno == socket.EAI_NONAME:
