#
# Do not use kernel keyring, it is not namespaced
#
--- /etc/krb5.conf	2016-08-29 19:09:04.000000000 +0000
+++ /etc/krb5.conf	2018-11-15 14:36:58.694929927 +0000
@@ -12,7 +12,7 @@
  forwardable = true
  rdns = false
 # default_realm = EXAMPLE.COM
- default_ccache_name = KEYRING:persistent:%{uid}
+# default_ccache_name = KEYRING:persistent:%{uid}
 
 [realms]
 # EXAMPLE.COM = {
#
# Prevent the default_ccache_name = KEYRING:persistent:%{uid} from
# being put back in during ipa-server-install/ipa-replica-install.
#
--- /usr/lib/python2.7/site-packages/ipapython/kernel_keyring.py       2017-10-18 13:41:28.000000000 +0000
+++ /usr/lib/python2.7/site-packages/ipapython/kernel_keyring.py       2018-09-19 06:54:09.654029061 +0000
@@ -62,7 +62,7 @@
     except ValueError:
         return False
 
-    return True
+    return False
 
 def has_key(key):
     """
