--- /usr/lib/python3.9/site-packages/ipaplatform/base/paths.py	2020-08-20 10:38:12.000000000 +0000
+++ /usr/lib/python3.9/site-packages/ipaplatform/base/paths.py	2020-08-24 18:31:19.978883075 +0000
@@ -53,7 +53,7 @@
     HTTPD_SSL_SITE_CONF = "/etc/httpd/conf.d/ssl.conf"
     HTTPD_CERT_FILE = "/var/lib/ipa/certs/httpd.crt"
     HTTPD_KEY_FILE = "/var/lib/ipa/private/httpd.key"
-    HTTPD_PASSWD_FILE_FMT = "/var/lib/ipa/passwds/{host}-443-RSA"
+    HTTPD_PASSWD_FILE_FMT = "/data/var/lib/ipa/passwds/{host}-443-RSA"
     # only used on Fedora
     HTTPD_IPA_WSGI_MODULES_CONF = None
     OLD_IPA_KEYTAB = "/etc/httpd/conf/ipa.keytab"
@@ -72,23 +72,23 @@
     IPA_NSSDB_DIR = "/etc/ipa/nssdb"
     IPA_NSSDB_PWDFILE_TXT = "/etc/ipa/nssdb/pwdfile.txt"
     COMMON_KRB5_CONF_DIR = "/etc/krb5.conf.d/"
-    KRB5_CONF = "/etc/krb5.conf"
+    KRB5_CONF = "/data/etc/krb5.conf"
     KRB5_FREEIPA = COMMON_KRB5_CONF_DIR + "freeipa"
     KRB5_FREEIPA_SERVER = COMMON_KRB5_CONF_DIR + "freeipa-server"
-    KRB5_KEYTAB = "/etc/krb5.keytab"
+    KRB5_KEYTAB = "/data/etc/krb5.keytab"
     LDAP_CONF = "/etc/ldap.conf"
     LIBNSS_LDAP_CONF = "/etc/libnss-ldap.conf"
     NAMED_CONF = "/etc/named.conf"
     NAMED_CONF_BAK = "/etc/named.conf.ipa-backup"
-    NAMED_CUSTOM_CONF = "/etc/named/ipa-ext.conf"
-    NAMED_CUSTOM_OPTIONS_CONF = "/etc/named/ipa-options-ext.conf"
+    NAMED_CUSTOM_CONF = "/data/etc/named/ipa-ext.conf"
+    NAMED_CUSTOM_OPTIONS_CONF = "/data/etc/named/ipa-options-ext.conf"
     NAMED_CONF_SRC = '/usr/share/ipa/bind.named.conf.template'
     NAMED_CUSTOM_CONF_SRC = '/usr/share/ipa/bind.ipa-ext.conf.template'
     NAMED_CUSTOM_OPTIONS_CONF_SRC = (
         '/usr/share/ipa/bind.ipa-options-ext.conf.template'
     )
     NAMED_VAR_DIR = "/var/named"
-    NAMED_KEYTAB = "/etc/named.keytab"
+    NAMED_KEYTAB = "/data/etc/named.keytab"
     NAMED_RFC1912_ZONES = "/etc/named.rfc1912.zones"
     NAMED_ROOT_KEY = "/etc/named.root.key"
     NAMED_MANAGED_KEYS_DIR = "/var/named/dynamic"
@@ -114,7 +114,7 @@
     CA_CERTIFICATES_BUNDLE_PEM = None
     CA_CERTIFICATES_DIR = None
     NSS_DB_DIR = "/etc/pki/nssdb"
-    PKI_CONFIGURATION = "/etc/pki"
+    PKI_CONFIGURATION = "/data/etc/pki"
     PKI_TOMCAT = "/etc/pki/pki-tomcat"
     PKI_TOMCAT_ALIAS_DIR = "/etc/pki/pki-tomcat/alias"
     PKI_TOMCAT_ALIAS_PWDFILE_TXT = "/etc/pki/pki-tomcat/alias/pwdfile.txt"
@@ -325,7 +325,7 @@
         "/var/lib/pki/pki-tomcat/ca/profiles/ca/caSignedLogCert.cfg")
     KRA_CS_CFG_PATH = "/var/lib/pki/pki-tomcat/conf/kra/CS.cfg"
     KRACERT_P12 = "/root/kracert.p12"
-    SAMBA_DIR = "/var/lib/samba"
+    SAMBA_DIR = "/data/var/lib/samba"
     SSSD_DB = "/var/lib/sss/db"
     SSSD_MC_GROUP = "/var/lib/sss/mc/group"
     SSSD_MC_PASSWD = "/var/lib/sss/mc/passwd"
--- /usr/lib/python3.9/site-packages/ipaplatform/redhat/paths.py	2019-11-19 09:36:59.087415395 +0000
+++ /usr/lib/python3.9/site-packages/ipaplatform/redhat/paths.py	2019-11-19 09:37:10.928527450 +0000
@@ -39,7 +39,7 @@
     AUTHCONFIG = '/usr/sbin/authconfig'
     AUTHSELECT = '/usr/bin/authselect'
     SYSCONF_NETWORK = '/etc/sysconfig/network'
-    NSSWITCH_CONF = '/etc/authselect/user-nsswitch.conf'
+    NSSWITCH_CONF = '/data/etc/authselect/user-nsswitch.conf'
 
 
 paths = RedHatPathNamespace()
#
# Prevent unneeded /etc/httpd/conf.modules.d/02-ipa-wsgi.conf from
# being created in runtime
#
--- /usr/lib/python3.9/site-packages/ipaplatform/fedora/paths.py.orig	2019-02-28 13:06:33.000000000 +0000
+++ /usr/lib/python3.9/site-packages/ipaplatform/fedora/paths.py	2019-03-04 09:01:28.728711145 +0000
@@ -30,9 +30,7 @@
 
 
 class FedoraPathNamespace(RedHatPathNamespace):
-    HTTPD_IPA_WSGI_MODULES_CONF = (
-        "/etc/httpd/conf.modules.d/02-ipa-wsgi.conf"
-    )
+    HTTPD_IPA_WSGI_MODULES_CONF = None
     NAMED_CRYPTO_POLICY_FILE = "/etc/crypto-policies/back-ends/bind.config"
     if HAS_NFS_CONF:
         SYSCONFIG_NFS = '/etc/nfs.conf'
--- /usr/lib/tmpfiles.d/var.conf	2018-10-29 00:59:14.000000000 +0000
+++ /usr/lib/tmpfiles.d/var.conf	2018-12-14 10:37:58.607898037 +0000
@@ -12,9 +12,9 @@
 L /var/run - - - - ../run
 
 d /var/log 0755 - - -
-f /var/log/wtmp 0664 root utmp -
-f /var/log/btmp 0660 root utmp -
-f /var/log/lastlog 0664 root utmp -
+L /var/log/wtmp - - - - /data/var/log/wtmp
+L /var/log/btmp - - - - /data/var/log/btmp
+L /var/log/lastlog - - - - /data/var/log/lastlog
 
 d /var/cache 0755 - - -

#
# Workaround https://github.com/freeipa/freeipa-container/issues/313
#
--- /usr/lib64/python3.9/shutil.py	2019-12-18 18:48:49.000000000 +0000
+++ /usr/lib64/python3.9/shutil.py	2020-03-11 16:17:24.727098610 +0000
@@ -160,6 +160,8 @@
                 raise
             return
         for name in names:
+            if name == 'security.selinux':
+                continue
             try:
                 value = os.getxattr(src, name, follow_symlinks=follow_symlinks)
                 os.setxattr(dst, name, value, follow_symlinks=follow_symlinks)
#
# rndc-confgen does not seem to write through the
# /etc/rndc.key -> /data/etc/rndc.key symlink
#
--- /usr/libexec/generate-rndc-key.sh	2020-05-26 08:23:25.000000000 +0000
+++ /usr/libexec/generate-rndc-key.sh	2020-06-04 12:07:40.032800430 +0000
@@ -15,18 +15,18 @@
 
 # This script generates /etc/rndc.key if doesn't exist AND if there is no rndc.conf
 
-if [ ! -s /etc/rndc.key -a ! -s /etc/rndc.conf ]; then
-  echo -n $"Generating /etc/rndc.key:"
-  if /usr/sbin/rndc-confgen -a -A hmac-sha256 -r /dev/urandom > /dev/null 2>&1
+if [ ! -s /data/etc/rndc.key -a ! -s /etc/rndc.conf ]; then
+  echo -n $"Generating /data/etc/rndc.key:"
+  if /usr/sbin/rndc-confgen -c /data/etc/rndc.key -a -A hmac-sha256 -r /dev/urandom > /dev/null 2>&1
   then
-    chmod 640 /etc/rndc.key
-    chown root:named /etc/rndc.key
-    [ -x /sbin/restorecon ] && /sbin/restorecon /etc/rndc.key
-    success $"/etc/rndc.key generation"
+    chmod 640 /data/etc/rndc.key
+    chown root:named /data/etc/rndc.key
+    [ -x /sbin/restorecon ] && /sbin/restorecon /data/etc/rndc.key
+    success $"/data/etc/rndc.key generation"
     echo
   else
     rc=$?
-    failure $"/etc/rndc.key generation"
+    failure $"/data/etc/rndc.key generation"
     echo
     exit $rc
   fi
