#
# Workaround https://pagure.io/freeipa/issue/8764
#
--- /usr/lib/python3.9/site-packages/ipaserver/install/installutils.py	2021-02-15 18:43:48.000000000 +0000
+++ /usr/lib/python3.9/site-packages/ipaserver/install/installutils.py	2021-03-19 15:50:08.068323156 +0000
@@ -1098,7 +1098,7 @@
             )
 
         with open(limit_file) as fd:
-            limit = fd.readline()
+            limit = fd.readline().rstrip()
         with open(usage_file) as fd:
             used = int(fd.readline())

