--- /usr/lib/python3.7/site-packages/ipaplatform/base/paths.py	2018-07-26 12:56:12.000000000 +0000
+++ /usr/lib/python3.7/site-packages/ipaplatform/base/paths.py	2018-09-12 10:09:32.450630323 +0000
@@ -74,14 +74,14 @@
     IPA_NSSDB_DIR = "/etc/ipa/nssdb"
     IPA_NSSDB_PWDFILE_TXT = "/etc/ipa/nssdb/pwdfile.txt"
     COMMON_KRB5_CONF_DIR = "/etc/krb5.conf.d/"
-    KRB5_CONF = "/etc/krb5.conf"
+    KRB5_CONF = "/data/etc/krb5.conf"
     KRB5_FREEIPA = COMMON_KRB5_CONF_DIR + "freeipa"
-    KRB5_KEYTAB = "/etc/krb5.keytab"
+    KRB5_KEYTAB = "/data/etc/krb5.keytab"
     LDAP_CONF = "/etc/ldap.conf"
     LIBNSS_LDAP_CONF = "/etc/libnss-ldap.conf"
     NAMED_CONF = "/etc/named.conf"
     NAMED_VAR_DIR = "/var/named"
-    NAMED_KEYTAB = "/etc/named.keytab"
+    NAMED_KEYTAB = "/data/etc/named.keytab"
     NAMED_RFC1912_ZONES = "/etc/named.rfc1912.zones"
     NAMED_ROOT_KEY = "/etc/named.root.key"
     NAMED_BINDKEYS_FILE = "/etc/named.iscdlv.key"
@@ -89,7 +89,7 @@
     NAMED_CRYPTO_POLICY_FILE = None
     NSLCD_CONF = "/etc/nslcd.conf"
     NSS_LDAP_CONF = "/etc/nss_ldap.conf"
-    NSSWITCH_CONF = "/etc/nsswitch.conf"
+    NSSWITCH_CONF = "/data/etc/nsswitch.conf"
     CHRONY_CONF = "/etc/chrony.conf"
     NTP_CONF = "/etc/ntp.conf"
     NTP_STEP_TICKERS = "/etc/ntp/step-tickers"
--- /usr/share/pki/server/etc/default.cfg	2018-08-13 23:33:33.000000000 +0000
+++ /usr/share/pki/server/etc/default.cfg	2018-09-12 10:21:02.634596425 +0000
@@ -203,7 +203,7 @@
 pki_source_subsystem_path=/usr/share/pki/%(pki_subsystem_type)s
 pki_path=/var/lib/pki
 pki_log_path=/var/log/pki
-pki_configuration_path=/etc/pki
+pki_configuration_path=/data/etc/pki
 pki_registry_path=/etc/sysconfig/pki
 pki_instance_path=%(pki_path)s/%(pki_instance_name)s
 pki_instance_log_path=%(pki_log_path)s/%(pki_instance_name)s
--- /usr/libexec/ipa/ipa-httpd-pwdreader	2018-10-05 18:30:34.000000000 +0000
+++ /usr/libexec/ipa/ipa-httpd-pwdreader	2018-11-16 07:17:55.235711545 +0000
@@ -13,7 +13,7 @@
 fi
 
 fname=${1/:/-}-$2
-pwdpath=/var/lib/ipa/passwds/$fname
+pwdpath=/data/var/lib/ipa/passwds/$fname
 
 # Make sure the values passed in do not contain path information
 checkpath=$(/usr/bin/realpath -e ${pwdpath} 2>/dev/null)
#
# Prevent unneeded /etc/httpd/conf.modules.d/02-ipa-wsgi.conf from
# being created in runtime
#
--- /usr/lib/python3.7/site-packages/ipaplatform/fedora/paths.py	2019-07-31 08:57:24.000000000 +0000
+++ /usr/lib/python3.7/site-packages/ipaplatform/fedora/paths.py	2019-08-15 10:22:04.801370731 +0000
@@ -30,9 +30,7 @@
 
 
 class FedoraPathNamespace(RedHatPathNamespace):
-    HTTPD_IPA_WSGI_MODULES_CONF = (
-        "/etc/httpd/conf.modules.d/02-ipa-wsgi.conf"
-    )
+    HTTPD_IPA_WSGI_MODULES_CONF = None
     NAMED_CRYPTO_POLICY_FILE = "/etc/crypto-policies/back-ends/bind.config"
     if HAS_NFS_CONF:
         SYSCONFIG_NFS = '/etc/nfs.conf'
--- /usr/lib/tmpfiles.d/var.conf	2018-10-29 00:59:14.000000000 +0000
+++ /usr/lib/tmpfiles.d/var.conf	2018-12-14 10:37:58.607898037 +0000
@@ -12,9 +12,9 @@
 L /var/run - - - - ../run
 
 d /var/log 0755 - - -
-f /var/log/wtmp 0664 root utmp -
-f /var/log/btmp 0660 root utmp -
-f /var/log/lastlog 0664 root utmp -
+L /var/log/wtmp - - - - /data/var/log/wtmp
+L /var/log/btmp - - - - /data/var/log/btmp
+L /var/log/lastlog - - - - /data/var/log/lastlog
 
 d /var/cache 0755 - - -

#
# Force writing journal to /run/log/journal which we symlink to
# /data/var/log/journal
#
--- /etc/systemd/journald.conf	2016-02-01 14:04:05.000000000 +0000
+++ /etc/systemd/journald.conf	2018-12-14 08:49:25.189295557 +0000
@@ -12,7 +12,7 @@
 # See journald.conf(5) for details.
 
 [Journal]
-#Storage=auto
+Storage=volatile
 #Compress=yes
 #Seal=yes
 #SplitMode=uid
#
# Support /var/lib/samba on /data volume
#
--- /etc/samba/smb.conf	2019-11-06 11:57:25.000000000 +0000
+++ /etc/samba/smb.conf	2019-11-19 08:54:22.264738866 +0000
@@ -4,6 +4,7 @@
 # you modified it.
 
 [global]
+	state directory = /data/var/lib/samba
 	workgroup = SAMBA
 	security = user
 
--- /usr/lib/python3.7/site-packages/ipaserver/install/adtrustinstance.py	2019-08-14 15:37:23.000000000 +0000
+++ /usr/lib/python3.7/site-packages/ipaserver/install/adtrustinstance.py	2019-11-19 08:34:09.065235181 +0000
@@ -465,7 +465,9 @@
         conf_fd.write('### Added by IPA Installer ###\n')
         conf_fd.write('[global]\n')
         conf_fd.write('debug pid = yes\n')
-        conf_fd.write('config backend = registry\n')
+        conf_fd.write('state directory = /data/var/lib/samba\n')
+        conf_fd.write('cache directory = /data/var/lib/samba\n')
+        conf_fd.write('include = registry\n')
         conf_fd.close()
 
     def __add_plugin_conf(self, name, plugin_cn, ldif_file):
#
# Workaround https://github.com/freeipa/freeipa-container/issues/313
#
--- /usr/lib64/python3.7/shutil.py	2019-12-18 18:48:49.000000000 +0000
+++ /usr/lib64/python3.7/shutil.py	2020-03-11 16:17:24.727098610 +0000
@@ -160,6 +160,8 @@
                 raise
             return
         for name in names:
+            if name == 'security.selinux':
+                continue
             try:
                 value = os.getxattr(src, name, follow_symlinks=follow_symlinks)
                 os.setxattr(dst, name, value, follow_symlinks=follow_symlinks)
