#
# Force systemd not to attempt to create session keyring
#
--- /dev/null  2018-09-15 15:00:07.785801994 +0000
+++ /usr/lib/systemd/system/pki-tomcatd@.service.d/nokeyring.conf      2018-09-15 15:07:41.939178255 +0000
@@ -0,0 +1,3 @@
+# Workaround https://github.com/systemd/systemd/issues/6281
+[Service]
+KeyringMode=inherit
--- /dev/null  2018-09-15 15:00:07.785801994 +0000
+++ /usr/lib/systemd/system/ipa-dnskeysyncd.service.d/nokeyring.conf   2018-09-15 15:07:41.939178255 +0000
@@ -0,0 +1,3 @@
+# Workaround https://github.com/systemd/systemd/issues/6281
+[Service]
+KeyringMode=inherit
