#
# Workaround 1364139
#
--- /usr/lib/python3.6/site-packages/ipaserver/install/server/replicainstall.py	2018-08-20 07:05:54.000000000 +0000
+++ /usr/lib/python3.6/site-packages/ipaserver/install/server/replicainstall.py	2018-09-17 22:26:51.954544756 +0000
@@ -732,7 +732,7 @@
     api.bootstrap(in_server=True, context='installer', confdir=paths.ETC_IPA)
     api.finalize()
 
-    installutils.verify_fqdn(config.master_host_name, options.no_host_dns)
+    installutils.verify_fqdn(config.master_host_name, options.no_host_dns, local_hostname=False)
 
     cafile = os.path.join(config.dir, "ca.crt")
     if not os.path.isfile(cafile):
@@ -1114,7 +1114,7 @@
                            "certificate")
 
     installutils.verify_fqdn(config.host_name, options.no_host_dns)
-    installutils.verify_fqdn(config.master_host_name, options.no_host_dns)
+    installutils.verify_fqdn(config.master_host_name, options.no_host_dns, local_hostname=False)
 
     ccache = os.environ['KRB5CCNAME']
     kinit_keytab('host/{env.host}@{env.realm}'.format(env=api.env),
