#
# Workaround 1364139
#
--- /usr/lib/python3.8/site-packages/ipaserver/install/server/replicainstall.py	2018-10-05 18:31:57.000000000 +0000
+++ /usr/lib/python3.8/site-packages/ipaserver/install/server/replicainstall.py	2018-11-08 13:45:48.223578505 +0000
@@ -877,7 +877,7 @@
                            "certificate")
 
     installutils.verify_fqdn(config.host_name, options.no_host_dns)
-    installutils.verify_fqdn(config.master_host_name, options.no_host_dns)
+    installutils.verify_fqdn(config.master_host_name, options.no_host_dns, local_hostname=False)
 
     ccache = os.environ['KRB5CCNAME']
     kinit_keytab('host/{env.host}@{env.realm}'.format(env=api.env),
